File: Kata 6/Buying a car/Buying a car.py
def nbMonths(startPriceOld, startPriceNew, savingperMonth, percentLossByMonth):
    currentPrice=startPriceOld
    month=0
    percent=1-percentLossByMonth*0.01
    
    while currentPrice < startPriceNew:
        month+=1           
        if month %2==0:
            percent=percent-0.5*0.01 
        startPriceOld*=percent 
        startPriceNew*=percent           
        currentPrice=savingperMonth*month+startPriceOld     
            
    return [month,round(currentPrice-startPriceNew)]

File: Kata 6/Buying a car/Buying a car.cs
using System;
public class BuyCar {

	public static int[] nbMonths(int startPriceOld, int startPriceNew, int savingPerMonth, double percentLossByMonth) {
		// your code
       double dcurrentPrice = startPriceOld;
       int month = 0;
       double dPriceNew = startPriceNew;
       double dPriceOld = startPriceOld;
       
       double percent = 1 - percentLossByMonth * 0.01;
       while (dcurrentPrice < dPriceNew)
       {
           month += 1;
           percent = month % 2 == 0 ? percent - 0.5 * 0.01 : percent;
           dPriceNew *= percent;
           dPriceOld *= percent;
           dcurrentPrice = savingPerMonth * month + dPriceOld;
       }
       return new int[] { month, (int)Math.Round(dcurrentPrice - dPriceNew) };
	}
}
